src/include/mit/mit.h.in

```diff
@@ -58,15 +52,44 @@ typedef uint64_t mit_UWORD;
 #define mit_UWORD_MAX UINT64_MAX
 #define mit_WORD_MIN INT64_MIN
 #define mit_WORD_MAX INT64_MAX
-#define PRI_WORD PRId64
-#define PRI_UWORD PRIu64
-#define PRI_XWORD "#"PRIx64
 #define reverse_endianness bswap_64
+#elif WORD_BYTES == 16
+/* Given an unsigned 128-bit argument X, return the value corresponding to
+   X with reversed byte order.  */
+#define bswap_128(x) ((((x) & (((unsigned __int128)0xFF << 0)))   << 120) | \
+                      (((x) & (((unsigned __int128)0xFF << 8)))   << 104) | \
+                      (((x) & (((unsigned __int128)0xFF << 16)))  << 88) | \
+                      (((x) & (((unsigned __int128)0xFF << 24)))  << 72) | \
+                      (((x) & (((unsigned __int128)0xFF << 32)))  << 56) | \
+                      (((x) & (((unsigned __int128)0xFF << 40)))  << 40) | \
+                      (((x) & (((unsigned __int128)0xFF << 48)))  << 24) | \
+                      (((x) & (((unsigned __int128)0xFF << 56)))  << 8) | \
+                      (((x) & (((unsigned __int128)0xFF << 64)))  >> 8) | \
+                      (((x) & (((unsigned __int128)0xFF << 72)))  >> 24) | \
+                      (((x) & (((unsigned __int128)0xFF << 80)))  >> 40) | \
+                      (((x) & (((unsigned __int128)0xFF << 88)))  >> 56) | \
+                      (((x) & (((unsigned __int128)0xFF << 96)))  >> 72) | \
+                      (((x) & (((unsigned __int128)0xFF << 104))) >> 88) | \
+                      (((x) & (((unsigned __int128)0xFF << 112))) >> 104) | \
+                      (((x) & (((unsigned __int128)0xFF << 120))) >> 120))
+
+typedef __int128 mit_WORD;
+typedef unsigned __int128 mit_UWORD;
+#define mit_SIZE_WORD 4
+#define mit_WORD_MASK ((__int128)-1)
+#define mit_UWORD_MAX mit_WORD_MASK
+#define mit_WORD_MIN (((unsigned __int128)1) << 127)
+#define mit_WORD_MAX ((((__int128)0x7fffffffffffffff) << 64) | 0xffffffffffffffff)
+#define reverse_endianness bswap_128
```

Comment:
UWORD_MAX should be cast to unsigned __int128.
WORD_MIN could be (__int128)(UWORD_MAX << 127).
WORD_MAX could be (__int128)(UWORD_MAX >> 1).

Reply:
Done, and more!